xds/src/main/java/io/grpc/xds/ClientLoadCounter.java

```diff
@@ -83,6 +83,8 @@ void recordMetric(String name, double value) {
   /**
    * Generates a snapshot for load stats recorded in this counter for the interval between calls
    * of this method.
+   *
+   * <p>Calls to this method must be serialized externally.
```

Comment:
What about declare it as a `synchronized` method instead of requiring serializing externally?

Reply:
Hmm... We can, but it doesn't need to.

Reply:
Okay, this is on the control path. So it doesn't matter much to use `synchronized`. I've made the `ClientLoadCounter` class completely thread-safe.